docs/grfn_spec.md

```diff
@@ -11,96 +20,189 @@ associated extracted comments, links to natural language text (the
+
+Examples:
+
+- `scope`: As will be described below, program analysis will assign unique names for scopes (see discussion below under conditional, container and loop\_plate functions). Given these names, the scope of the inner loop within the function `foo` in this example,
+    ```
+    def foo():
+        for i in range(10):      # assigned name 'loop$1'
+            for j in range(10):  # assigned name 'loop$1' (in the scope of the outer loop$1)
+                x = i*j
+    ```
+    would be uniquely specified by the following path:
+    ```
+    ["foo", "loop$1", "loop$1"]
+    ```
+    
+    In general, it is not necessary within GrFN to independently declare `scope`s. Instead, we simply specify the `<scope_path>` in an indicator declaration as a list of strings under the "scope" attribute in the identifier declaration (below).
+    
+    ```
+    <scope_path> ::= list of <string>
+    ```
+    
+- `namespace`: Different languages have different conventions for defining namespaces, but in general they are either (1) explicitly defined within source code by namespace declarations (such as Fortran "modules" or C++ "namespace"s), or (2) implicitly defined by the project directory structure within which a file is located (as in Python). In the case of namespaces defined by project directory structure, two files in different locations in the project directory tree may have the same name. To distinguish these, program analysis will capture the path of the directory tree from the root to the file. The final name in the path, which is the name of the source file, will drop the file extension. For example, the namespace for file `baz.py` within the following directory tree
+    ```
+    foo/
+        bar/
+            baz.py
+    ```
+    would be the uniquely specified by the following path:
+    ```
+    ["foo", "bar", "baz"]
+    ```
+
+	In the case of declared namespaces, the namespace declaration will determine the path (which may only consist of one string name).
+
+    Again, it is not necessary within GrFN to independently declare a `namesapce`; like the `<scope_path>`, we specify the `<namespace_path>` within an identifier declaration as a list strings under the "namespace" attribute in the identifie declaration:
```

Comment:
`namesapce` --> `namespace`

attribute in the identifie declaration --> attribute in the identifier declaration